projects/hipdnn/sdk/include/hipdnn_sdk/utilities/LoadGraphAndTensors.hpp

```
@@ -0,0 +1,246 @@
+            if(!passedValidation)
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    std::unordered_map<int64_t, void*> hostBufferMap()
+    {
+        std::unordered_map<int64_t, void*> bufferMap;
+        for(auto& [uid, tensor] : tensorMap)
+        {
+            bufferMap[uid] = tensor->rawHostData();
+        }
+
+        return bufferMap;
+    }
+};
+
+inline std::vector<int64_t> getOutputTensorUidsFromGraph(nlohmann::json graph)
+{
+    std::vector<int64_t> outputTensorUids;
+
+    for(auto const& node : graph.at("nodes"))
+    {
+        for(auto& [name, value] : node.at("outputs").items())
+        {
+            if(name.find("_tensor_uid") == std::string::npos)
+            {
+                continue;
+            }
+
+            outputTensorUids.push_back(value.get<int64_t>());
+        }
+    }
+
+    return outputTensorUids;
+}
+
+inline GraphAndTensorMap loadGraphAndTensors(std::filesystem::path const& path
+                                             /*bool separateOutputs = false*/)
+{
+    auto basePath = path;
+    basePath.replace_extension();
+
+    nlohmann::json graphJson = [](auto const& path) {
+        std::ifstream graphFileStream(path);
+        if(!graphFileStream)
+        {
+            throw std::runtime_error("Error in loadGraphAndTensors(): file could not be opened "
+                                     + path.string());
+        }
+        return nlohmann::json::parse(graphFileStream);
+    }(path);
+
+    flatbuffers::FlatBufferBuilder graphBuilder;
+    auto graphOffset
+        = hipdnn_sdk::json::to<hipdnn_sdk::data_objects::Graph>(graphBuilder, graphJson);
+    graphBuilder.Finish(graphOffset);
+
+    auto graph = data_objects::GetGraph(graphBuilder.GetBufferPointer());
+
+    auto outputTensorUids = getOutputTensorUidsFromGraph(graphJson);
+
+    std::unordered_map<int64_t, std::unique_ptr<ITensor>> tensorMap;
+    for(auto attributes : *graph->tensors())
```

I think we need to check graph tensors before deref.
It would be wrong if tensors was empty anyways, but probably better to get a typed error instead of null ptr deref.

---

Would you ever have a graph with no tensors? I'm debating whether this should throw an error or if we just end up with an empty tensor map

---

I opted to throw, at least currently I believe every node requires at least one tensor

---

Throw makes sense to me.